### PR TITLE
doc: update architecture design doc and graph

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

You can preview the change [here](https://github.com/risingwavelabs/risingwave/blob/3050f8d60ca7b77126e583d0df0422e6e97bee07/docs/architecture-design.md)

- Update architecture design doc to include compactor node description.
- Prettify architecture graph.

TODO: I will make the original SVGs in our repo more friendly to dark-theme by adding a while background in the next PR.

### CODE REVIEW

File: docs/architecture-design.md [hunk 1]
```diff
@@ -6,11 +6,18 @@ This document serves as one of the materials for newcomers to learn the high-lev
...
 
-There are currently 3 types of nodes in the cluster: 
+There are currently 4 types of nodes in the cluster: 
 
 * **Frontend**: Frontend is a stateless proxy that accepts user queries through Postgres protocol. It is responsible for parsing, validation, optimization, and answering the results of each individual query. 
 * **ComputeNode**: ComputeNode is responsible for executing the optimized query plan. 
-* **MetaServer**: The central metadata management service. It also acts as a failure detector that periodically sends heartbeats to frontends and compute-nodes in the cluster.
+* **Compactor**: Compactor is a stateless worker node responsible for executing the compaction tasks for our storage engine.
+* **MetaServer**: The central metadata management service. It also acts as a failure detector that periodically sends heartbeats to frontends and compute-nodes in the cluster. There are multiple sub-components running in MetaServer:
+   * **ClusterManager**: Manage the
```
Comment (neverchanje) [inline docs/architecture-design.md:15]: Manage the cluster information.

Comment (hzxa21) [reply docs/architecture-design.md]: Fixed.

**Review by arkbriar — APPROVED** [review]
LGTM! Thanks a lot! 

cc @wyhyhyhyh

**Comment by wyhyhyhyh** [comment]
LGTM!

**Review by wyhyhyhyh — APPROVED** [review]

**Pull request merged**